[#271] Restricted variable names with more than one consecutive underscore

Roxe322 opened the pull request:
# I have made things!

## Checklist

<!-- Please check everything that applies: -->

- [V] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [V] I have created at least one test case for the changes I have made
- [V] I have updated the documentation for the changes I have made
- [V] I have added my changes to the `CHANGELOG.md`

## Related issues

- Closes #271

Roxe322 commented:
@sobolevn I'm really not sure that I've made some right things. First is moving `logics/variable` to module and second is my test (new file and breaking DRY in it). Please, help me with solving it, because I'm really afraid that my decisions in this was wrong

Roxe322 commented:
Also, I've just noticed that `pyproject.lock` was deleted. How can I resolve it? Probably it's because I've installed it with last version of poetry and now I have `poetry.lock` file instead of that.

sobolevn commented on wemake_python_styleguide/logics/variables/name_helpers.py:
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+import ast
```
Missing empty line after `# -*-`

sobolevn commented on wemake_python_styleguide/violations/naming.py:
```diff
@@ -521,3 +524,28 @@ class A(object):
+        # Correct:
+        some_value = 5
+        __magic__ = 5
+        # Wrong:
```
There should be an empty line here

sobolevn commented on wemake_python_styleguide/violations/naming.py:
```diff
@@ -521,3 +524,28 @@ class A(object):
+        This naming rule already exist for module names.
+
+    Example::
+        # Correct:
```
There should be an empty line here before `# Correct:`

sobolevn commented on wemake_python_styleguide/violations/naming.py:
```diff
@@ -521,3 +524,28 @@ class A(object):
+        __magic__ = 5
+        # Wrong:
+        some__value = 5
+    Note:
```
There should be an empty line here before `Note:`

sobolevn commented:
@Roxe322 considering your thoughts about `dry` inside the tests.
Yes, this is a problem. But, we are going to solve it separately.

Roxe322 commented:
Hi, @sobolevn , I've fixed missing blank lines and renamed modules as you suggested. What should I do with DRY problem in tests? Also, I've just noticed that Travis broke in 3.7 tests. Why such problem occured and why in 3.6 everything is ok? I'm using 3.7 on my local machine and pytest working normally.

sobolevn commented:
You have the same problem as https://github.com/wemake-services/wemake-python-styleguide/pull/280

I am not sure what happens here.

vsmaxim commented:
Hi, @Roxe322, travis ci is failing due to wrong tests. For more info check this: #283.

Roxe322 commented:
Thank you, @vsmaxim. I'll wait @sobolevn to merge your PR, than pull it and update mine :)

Roxe322 commented:
Thank you, @vsmaxim. I'll wait @sobolevn to merge your PR, than pull it and update mine :)

Roxe322 commented:
Thank you, @vsmaxim. I'll wait @sobolevn to merge your PR, than pull it and update mine :)

sobolevn commented:
@Roxe322 can you please `.. versionadded:: 0.3.0` to you violation docs? And then I will merge it.

Roxe322 commented:
@sobolevn Ok, also i'll fix merge conflicts

sobolevn commented:
@Roxe322 please, check your build status.

Roxe322 commented:
@sobolevn Yes, I'm working on this, just not very successful merge conflicts resolving

sobolevn commented:
Awesome! Thanks!
